ambry-store/src/main/java/com/github/ambry/store/CompactionLog.java

```diff
@@ -128,6 +123,47 @@ static void processCompactionLogs(String dir, String storeId, StoreKeyFactory st
     }
   }
 
+  /**
+   * Remove all the compaction files that are started before the given {@code beforeTime}.
+   * @param dir The data dir that contains all the compaction log files.
+   * @param storeId The store id.
+   * @param beforeTime The cutoff time. Compaction log file started before this time will be removed.
+   */
+  static void cleanupCompactionLogs(String dir, String storeId, long beforeTime) {
+    List<File> sortedFiles = getCompactedLogFilesInDescendingOrder(dir, storeId);
+    if (sortedFiles.isEmpty()) {
+      logger.info("{}: No compaction log to cleanup", storeId);
+      return;
+    }
+    // The list is in descending order, the most recent compaction logs are in the front, we have to reverse the list
+    for (int i = sortedFiles.size() - 1; i >= 0; i--) {
+      File compactionFile = sortedFiles.get(i);
+      long startTime = getStartTimeFromFile(compactionFile);
+      if (startTime >= beforeTime) {
+        break;
+      } else {
+        logger.info("{}: Removing compaction file: {}, cutoff time: {}", storeId, compactionFile.getName(), beforeTime);
+        compactionFile.delete();
```

Comment:
Is there any change we are still writing the compaction file? Is there any way to check for that?

Reply:
the in progress compaction file is not included in the filter, so it won't even show up in the list.